### ISSUE TITLE
UNIT.getVanillaUnitType

### ISSUE THREAD

**Issue by FranticFairy** [Robosturm/Commander_Wars#458]
In developing Brave New World, I have encountered an issue.

Through adding new UnitTypes, I have broken compatibility with COs, Weather, and anything else that asks for UnitType, whether it be in the base game or in other mods. And the only ways I can see to fix this is either refraining from adding more UnitTypes, or making every other modder add some more code to their mod.

Therefore, I am proposing a function that should be used instead of getUnitType() for when the UnitType can be a bit more generic, but not as generic as unitTypeToGround():

```
UNIT.getVanillaUnitType = function(unitType) {
    if (
        unitType === GameEnums.UnitType_Infantry
    ) {
        return GameEnums.UnitType_Infantry;
    } else if (
        unitType === GameEnums.UnitType_Ground
    ) {
        return GameEnums.UnitType_Ground;
    } else if (
        unitType === GameEnums.UnitType_Hovercraft
    ) {
        return GameEnums.UnitType_Hovercraft;
    } else if (
        unitType === GameEnums.UnitType_Air
    ) {
        return GameEnums.UnitType_Air;
    } else if (
        unitType === GameEnums.UnitType_Naval
    ) {
        return GameEnums.UnitType_Naval;
    }
    return unitType;
}
```

Without mods, this function does the exact same thing as getUnitType(). However, with mods, it would look something like this:

```
UNIT.getVanillaUnitType = function(unitType) {
    if (
        unitType === GameEnums.UnitType_Infantry ||
        unitType === GameEnums.BNW_Infantry ||
        unitType === GameEnums.BNW_FieldGun
    ) {
        return GameEnums.UnitType_Infantry;

    } else if (
        unitType === GameEnums.UnitType_Ground ||
        unitType === GameEnums.BNW_Vehicle ||
        unitType === GameEnums.BNW_Support ||
        unitType === GameEnums.BNW_Tank ||
        unitType === GameEnums.BNW_Train
    ) {
        return GameEnums.UnitType_Ground;

    } else if (
        unitType === GameEnums.UnitType_Hovercraft
    ) {
        return GameEnums.UnitType_Hovercraft;

    } else if (
        unitType === GameEnums.UnitType_Air ||

        unitType === GameEnums.BNW_FPlane ||
        unitType === GameEnums.BNW_SPlane ||
        unitType === GameEnums.BNW_UPlane
    ) {
        return GameEnums.UnitType_Air;

    } else if (
        unitType === GameEnums.UnitType_Naval ||

        unitType === GameEnums.BNW_SShip ||
        unitType === GameEnums.BNW_HShip ||
        unitType === GameEnums.BNW_UShip
    ) {
        return GameEnums.UnitType_Naval;
    }
    return unitType;
}
```

In this situation, it would return what the modder determines is the most accurate Vanilla UnitType for their units, if any. This way, support for new UnitTypes can be handled by the person adding new UnitTypes.

tl;dr: Three functions:
getUnitType() -> Returns the _exact_ unitType - not Compatible with new UnitTypes
unitTypeToGround() -> Returns _Ground_, _Air_, or _Naval_ - not Specific enough
getVanillaUnitType() -> For turning modded UnitTypes into vanilla UnitTypes

**Comment by Robosturm** [comment]
So why i am denying this:
1. From what i read you want to have some sort of sub-unit types, but your inverting the logic/dependency with this approach
2. Since you introduce some sort of infantry subtypes for example it's more on the mods shoulder to introduce them rather than a base game thing since the base game doesn't have those sub-types.
3. You can add a function getUnitSubtype to your modded units and check for sad subtype when you need it. --> The downside of this would be a potential less organized editor selection#

E.g.:
in the unit.js file you add do something or even give the unit as additional argument so you can do some more generic interpetation.
 like UNIT.getUnitSubtype() = function()
{
   return GameEnums.UnitType_Naval; 
}
